Fabric/src/main/java/com/techarium/techarium/mixin/RegistryAccessMixin.java

```diff
@@ -0,0 +1,23 @@
+package com.techarium.techarium.mixin;
+
+import com.google.common.collect.ImmutableMap;
+import com.techarium.techarium.multiblock.MultiBlockStructure;
+import com.techarium.techarium.platform.FabricRegistryHelper;
+import net.minecraft.core.Registry;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.resources.ResourceKey;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+
+@Mixin(RegistryAccess.class)
+public interface RegistryAccessMixin {
+
+	@ModifyVariable(method = "method_30531()Lcom/google/common/collect/ImmutableMap;", at = @At("STORE"), ordinal = 0)
+	private static ImmutableMap.Builder<ResourceKey<? extends Registry<?>>, RegistryAccess.RegistryData<?>> swapMapV2(ImmutableMap.Builder<ResourceKey<? extends Registry<?>>, RegistryAccess.RegistryData<?>> builder) {
```

Comment:
This seems pretty prone to break, especially if another mod does a similar thing. I'm not sure if we have a way around that, but I do feel that should be discussed.

Reply:
I believe you can use the mixin to inject into the lambda before the return, and then use capture locals to get access the builder and modify it.

Reply:
ModifyVariable can not be chained afaik, if multiple mods do the same, there will be a crash.